src/core_atmosphere/mpas_atm_core_interface.F

```diff
@@ -333,19 +333,178 @@ end function atm_setup_decompositions
+      ! are to be allocated.
+      !
+      if (associated(block % domain % configs)) then
+          nullify(cam_pcnst)
+          err_level = mpas_pool_get_error_level()
+          call mpas_pool_set_error_level(MPAS_POOL_SILENT)
+          call mpas_pool_get_config(block % domain % configs, 'cam_pcnst', cam_pcnst)
+          call mpas_pool_set_error_level(err_level)
+          if (associated(cam_pcnst)) then
+             call mpas_log_write('')
+             call mpas_log_write('** Config ''cam_pcnst'' is defined with a value of $i', intArgs=[cam_pcnst])
+             call mpas_log_write('   Scalars will be allocated separately from Registry-defined variables')
+             call mpas_log_write('')
+             ierr = atm_allocate_scalars(block, cam_pcnst)
+          end if
+      end if
+
    end function atm_setup_block
 
 
+   !***********************************************************************
+   !
+   !  function atm_allocate_scalars
+   !
+   !> \brief   Allocate scalars and scalars_tend var_arrays
+   !> \author  Michael G. Duda
+   !> \date    20 May 2020
+   !> \details 
+   !>  When MPAS-A is operating as a dycore for CAM, the scalars and
+   !>  scalars_tend var_arrays are allocated separately from other Registry-
+   !>  defined variables, since the set of scalars to be handled by the dycore
+   !>  is not known until runtime. This routine allocates these var_arrays,
+   !>  but it does not define which constituent is at which position in
+   !>  var_arrays; this is defined later in the CAM-MPAS interface layer.
+   !
+   !-----------------------------------------------------------------------
+   function atm_allocate_scalars(block, num_scalars) result(ierr)
+
+      use mpas_derived_types, only : block_type
+
+      use mpas_derived_types, only : mpas_pool_type, field3dReal, MPAS_LOG_ERR
+      use mpas_pool_routines, only : mpas_pool_get_subpool, mpas_pool_add_dimension, mpas_pool_add_field
+      use mpas_log, only : mpas_log_write
+
+      implicit none
+
+      ! Arguments
+      type (block_type), pointer :: block
+      integer, intent(in) :: num_scalars
+
+      ! Return value
+      integer :: ierr
+
+      ! Local variables
+      integer :: i, j, timeLevs
+      type (mpas_pool_type), pointer :: statePool
+      type (mpas_pool_type), pointer :: tendPool
+      type (field3dReal), dimension(:), pointer :: scalarsField
+
+
+      ierr = 0
+
+      !
+      ! Allocate scalars var_array
+      !
+      nullify(statePool)
+      call mpas_pool_get_subpool(block % structs, 'state', statePool)
+
+      if (.not. associated(statePool)) then
+         call mpas_log_write('No pool named ''state'' was found in atm_allocate_scalars', messageType=MPAS_LOG_ERR)
+         ierr = 1
+         return
+      end if
+
+      timeLevs = 2
+
+      call mpas_pool_add_dimension(statePool, 'num_scalars', num_scalars)
+
+      allocate(scalarsField(timeLevs))
+
+      do i = 1, timeLevs
```

Comment:
Can you explain to me the need to loop over `timeLevs` when creating  the `scalarsField`?

Reply:
Since the state variables have more than one time level (see the definition of the [state var_array](https://github.com/MPAS-Dev/MPAS-Model/blob/v7.0/src/core_atmosphere/Registry.xml#L1412)), we need to initialize all time levels of the scalars.